### ISSUE TITLE
Advanced Blocking - TXT file with a collection of URLs not working

### ISSUE THREAD

**Issue by dannykorpan** [TechnitiumSoftware/DnsServer#600]
Hi,

if I define a URL to a TXT blocking list which includes a collection of blocking lists, blocking is not working e.g. with this list https://v.firebog.net/hosts/lists.php. But in Settings -> Blocking -> Allow/Block List URLs it's working.

Kind Regards,
Danny

**Comment by ShreyasZare** [comment]
Thanks for the post. From what I understand, you configured a URL which returns a list of block list URLs? In that case its not going to work. The DNS Server is expecting a URL that returns list of domain names to block.

**Comment by dannykorpan** [comment]
That's correct. But within the Settings -> Blocking are it's working. Otherweise it's a suggestion for improvement.

**Comment by ShreyasZare** [comment]
You can just copy those URLs and paste them in the settings directly. How will this feature help since its not like you need to change the block list URLs frequently?

**Comment by ShreyasZare** [comment]
Like what @BonnieSoftware said about inconsistency being one issue, the other issue is that it can cause cyclic dependency like a list of URLs where one of the URL returns a list of URLs.

For such specific requirements, its best that you write a small bash script that uses curl to call the HTTP API and update the block list URLs. You can then run this script with cron and it will keep it updated automatically.

**Comment by dannykorpan** [comment]
Thank you for the tip, I think there is still room for improvement. For a URL where one of the URL returns a list of URLs could be an acceptable restriction.

**Comment by ShreyasZare** [comment]
> Thank you for the tip, I think there is still room for improvement. For a URL where one of the URL returns a list of URLs could be an acceptable restriction.

This is a very specific requirement which is not commonly needed and so is not justifying having a full feature support. Give a try with bash script, it will be quite easy to get it to work.

**Comment by dannykorpan** [comment]
A bash script is totally fine for me, but possibly not for others. If you want a sustainable product with included blocking feature, it's a must have function.

**Comment by ShreyasZare** [comment]
Ok so lets keep this idea open for a while which will allow more time to think on it.
